### PR TITLE
django-selectable ->  django-autocomplete-light

### PR DESCRIPTION
**Pull request by rabstejnek**
Replace [django-selectable](https://github.com/mlavin/django-selectable) (no longer supported) with [django-autocomplete-light](https://django-autocomplete-light.readthedocs.io/en/master/).

In addition:

- remove [jquery UI](https://jqueryui.com/)
- update [htmx] (https://github.com/bigskysoftware/htmx/blob/master/CHANGELOG.md) 1.6.1 -> 1.8.0
- write a new client-side autocomplete method now that jquery UI is removed

### CODE REVIEW

File: hawc/apps/myuser/autocomplete.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from ..common.autocomplete import BaseAutocomplete, registry
+from . import models
+
+
+class UserAutocomplete(BaseAutocomplete):
+    model = models.HAWCUser
+    search_fields = ["first_name", "last_name"]
+
+
+registry.register(UserAutocomplete)
```
Comment (shapiromatron) [inline hawc/apps/myuser/autocomplete.py:10]: can we make this a class decorator instead?

File: hawc/apps/common/autocomplete.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+from dal import autocomplete
+from django.http import Http404
+from django.utils.encoding import force_str
+
+from .helper import reverse_with_query_lazy
+
+
+class BaseAutocomplete(autocomplete.Select2QuerySetView):
+    # AJAX only? `request.is_ajax()`
```
Comment (shapiromatron) [inline hawc/apps/common/autocomplete.py:9]: yes, let's confirm it's ajax

File: hawc/apps/common/autocomplete.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    # Other permissions?
```
Comment (shapiromatron) [inline hawc/apps/common/autocomplete.py:10]: i think we should by default require user is authenticated too, though i think in some cases we may need to add additional permissions

File: hawc/apps/common/autocomplete.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    @classmethod
+    def name(cls):
```
Comment (shapiromatron) [inline hawc/apps/common/autocomplete.py:13]: rename to `registry_key`?

File: hawc/apps/common/helper.py [hunk 5]
```diff
@@ -411,3 +414,13 @@ def event_plot(series: pd.Series) -> Axes:
 
     plt.tight_layout()
     return ax
+
+
+def reverse_with_query(*args, query: dict, **kwargs):
```
Comment (shapiromatron) [inline hawc/apps/common/helper.py:419]: add docstring

File: hawc/apps/assessment/forms.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 from pathlib import Path
```
Comment (shapiromatron) [inline hawc/apps/assessment/forms.py:1]: changes in this file seem appropriate. I'd recommend removing for this PR. For future work, we'll probably go app by app and replace the existing functionality and replace with the new

File: hawc/apps/assessment/forms.py [hunk 7]
```diff
@@ -44,6 +46,10 @@ class Meta:
         model = models.Assessment
         widgets = {
             "public_on": DateCheckboxInput,
+            "project_manager": autocomplete.ModelSelect2Multiple(
```
Comment (shapiromatron) [inline hawc/apps/assessment/forms.py:49]: We'll probably want to build a helper method or class to specify the bootstrap theme by default. But that may come when we start implementing

File: hawc/main/settings/base.py [hunk 8]
```diff
@@ -93,6 +93,8 @@
     "django.contrib.sites",
     "django.contrib.messages",
     "django.contrib.staticfiles",
+    "dal",
```
Comment (shapiromatron) [inline hawc/main/settings/base.py:96]: move these into the "external apps" section; honestly while we're at it we should move django_filters down too

File: hawc/templates/base.html [hunk 9]
```diff
@@ -108,5 +108,6 @@
     <script type="text/javascript" src="{% static 'vendor/htmx/1.6.1/htmx.min.js' %}"></script>
     {% render_bundle 'main' %}
     {% block extrajs %}{% endblock %}
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/select2-bootstrap-theme/0.1.0-beta.10/select2-bootstrap.min.css">
```
Comment (shapiromatron) [inline hawc/templates/base.html:111]: add this to our vendored files in static/css: https://cdnjs.com/libraries/select2-bootstrap-theme

File: tests/hawc/apps/assessment/test_view_permissions.py [hunk 10]
```diff
@@ -15,8 +15,7 @@
     "noel_name": 0,
     "rob_name": 0,
     "editable": "on",
-    "project_manager_0": ("2",),
-    "project_manager_1": ("2",),
+    "project_manager": ("2",),
```
Comment (shapiromatron) [inline tests/hawc/apps/assessment/test_view_permissions.py:18]: i guess we can revert this change too, will occur when we actually roll this out

**Review by shapiromatron — CHANGES_REQUESTED** [review]
This is great work. After we complete laying the framework here, we'll probably progressively work through the app, replacing one `lookups.py` file at at a time...

File: hawc/main/settings/base.py (hunk 8 above)
Comment (rabstejnek) [reply hawc/main/settings/base.py:96]: It's suggested to include these before `django.contrib.admin`, which is why its up here and not in the "external apps" section:
https://django-autocomplete-light.readthedocs.io/en/master/install.html#configuration

**Comment by rabstejnek** [comment]
@shapiromatron ready for review!

**Review by shapiromatron — APPROVED** [review]
Excited to merge this in! Really big and complicated PR, the autocomplete abstractions look very convenient and easy to use. I changed the single file into a package to give it some room to grow, and made a few other minor changes, but the bones of this PR were solid and clearly very well tested and designed.

Excellent job! Goodbye django-selectable, you've served us well (and also jquery UI).

We will have to figure out how to deal with this flaky epiv2 integration test a little further down the road...
